lib/libc/stdio.c

```diff
@@ -24,9 +34,137 @@ FILE __stdio_FILEs[3] = {
 };
 #undef DEFINE_STDIO_DESC
 
+FILE *fopen(const char *filename, const char *mode) {
+#ifdef WITH_LIB_FS
+    FILE *stream = (FILE *) malloc(sizeof(FILE));
+    if (stream == NULL) {
+        return NULL;
+    }
+
+    stream->use_fs = true;
+    stream->fs_handle.offset = 0;
+    stream->fs_handle.readonly = (!strchr(mode, 'w') && !strchr(mode, 'a'));
+
+    status_t ret = fs_open_file(filename, &(stream->fs_handle.handle));
+
+    if (ret == ERR_NOT_FOUND && !stream->fs_handle.readonly) {
+        ret = fs_create_file(filename, &(stream->fs_handle.handle), 0);
+    }
+
+    if (ret != NO_ERROR) {
+        free(stream);
+        return NULL;
+    }
+
+    if (strchr(mode, 'a')) {
+        struct file_stat stat;
+        if (NO_ERROR == fs_stat_file(stream->fs_handle.handle, &stat)) {
+            stream->fs_handle.offset = stat.size;
+        }
+    }
+
+    return stream;
+#else
+    return NULL;
+#endif // WITH_LIB_FS
+}
+
+int fclose(FILE *stream) {
+#ifdef WITH_LIB_FS
+    if (stream && !stream->use_fs) {
+        fs_close_file(stream->fs_handle.handle);
+        free(stream);
+    }
+    return 0;
+#else
+    return 0;
+#endif // WITH_LIB_FS
+}
+
+size_t fread(void *ptr, size_t size, size_t count, FILE *stream) {
+#ifdef WITH_LIB_FS
```

Comment:
This will probably be a little cleaner to reorder such that the io_read part is the same, but it jsut #ifs in the (stream->use_fs) { } part.